docs/specs/code/TRANSACTION_MEMO.md

```diff
@@ -0,0 +1,87 @@
+
+A new textarea was added to `/components/wallet/send/WalletSendForm.js`
+
+### Flow
+
+The user flow is the following:
+1. User is going to make a transaction. He is looking at `/components/wallet/send/WalletSendForm.js`.
+2. The memo textarea is closed by default. User can click "Add memo" label.
+3. If user clicked "Add memo", he or she can see the following:
+   - If user haven't synced his external storage account yet, he or she will see the text "In order to attach private memos to   transactions you need to link Dropbox account." and a link to AccountsSettingsPage.
+     - If user clicks the link, he or she will be redirected to AccountsSettingsPage. Here user can click "Authorize Yoroi in    Dropbox". If user does, he or she will be asked to authorize Yoroi in Dropbox. After that user will be redirected to a    home page.
+   - If user have already synced his external storage account, he or she will see an opened textarea where the user can enter    the memo.
+4. If user clicks "Add memo" again:
+   - If the textarea is empty, it will be just closed.
+   - If it isn't empty, user will see the `/components/wallet/send/WalletCloseMemoDialog.js` dialog window with the text "Remove memo?" and two buttons, "Yes" and "No".
+     - If user clicks "No", the dialog closes and the memo textarea remain opened and have as content user's memo.
+     - If user clicks "Yes", the dialog window closes and the textarea become cleared and closes too.
+5. If user have entered the memo and makes a transaction, memo will be saved to localStorage and also to user's external storage.
+6. When user goes to his summary - `/containers/wallet/WalletSummaryPage.js` and clicks on the transaction to expand it, he or she will see his memo at the very bottom of Transaction item (if memo exists).
+7. When user links his external storage account to Yoroi, we check if there are some memos in his external storage. If there are, we download a content of each memo file via Dropbox API, read the content with FileReader object, then we save these memos to localStorage. In this way we can make user be able to install Yoroi on the another device, link his or her external storage account and get user's memos so they can be displayed in WalletSummary.
+
+### Tech spec
+
+#### External storages
+
+We use official [Dropbox Javascript SDK](https://github.com/dropbox/dropbox-sdk-js#readme) for authenticating and making requests to Dropbox API. We can create folders, files, read them and change the content.
+
+To make us able to auth user and manipulate his or her files, we need to [create an app in Dropbox](https://www.dropbox.com/developers/apps/create?_tk=pilot_lp&_ad=topbar5&_camp=create). When we create an app, we can get a `client_id` and we must provide a `redirect_uri` because Dropbox uses Oauth2. Both `client_id` and `redirect_uri` are used to create a link for user, which user should click to authenticate.
```

Comment:
Looks like Dropbox uses an app-user pair for rate limiting so it should be okay to register our app and not worry 👍